docs/go-configuration.md

```diff
@@ -4,27 +4,599 @@ title: Scope Go Agent configuration
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithRetriesOnFail(3)))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Handling test panic as a failure
+
+By default, the standard testing framework exits the test process on panic. This behavior can be changed to handle panics as test failures, allowing the complete execution of the test suite. 
+
+Also, this option can be combined with test retries, to retry tests that panic.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_TESTING_PANIC_AS_FAIL=true
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithHandlePanicAsFail()))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithHandlePanicAsFail()))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Adding agent metadata
+
+You can add custom agent metadata that will be appended to the default metadata that the agent collects (such as host information).
+
+Additionally, it is possible to set an environment variable as metadata value for a certain key, whose final value will be evaluated at runtime.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_METADATA="CustomKey1=MyValue, CustomKey2=MyValue2, CustomPath=\$PATH"
+```
+
+This is equivalent to: 
+```go
+    map[string]interface{}{
+		"CustomKey1": "MyValue",
+		"CustomKey2": "MyValue2",
+		"CustomPath": os.Getenv("PATH"),
+	}
+```
+
+<!--Testing instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithMetadata(map[string]interface{}{
+        "CustomKey1": "MyValue",
+        "CustomKey2": "MyValue2",
+        "CustomPath": os.Getenv("PATH"),
+    })))
+}
+
+func TestExample(t *testing.T) {
+    // ...
+}
+
+func BenchmarkExample(b *testing.B) {
+    // ...
+}
+```
+
+<!--Runtime instrumentation-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithMetadata(map[string]interface{}{
+		"CustomKey1": "MyValue",
+		"CustomKey2": "MyValue2",
+		"CustomPath": os.Getenv("PATH"),
+	})))
```

Comment:
This doesn't seem like runtime instrumentation

Reply:
ok